src/Illuminate/Console/Scheduling/Schedule.php

```diff
@@ -146,23 +146,19 @@ public function command($command, array $parameters = [])
 
     /**
      * Add a new job callback event to the schedule.
-     *
-     * @param  object|string  $job
-     * @param  string|null  $queue
-     * @param  string|null  $connection
-     * @return \Illuminate\Console\Scheduling\CallbackEvent
      */
-    public function job($job, $queue = null, $connection = null)
+    public function job(object|string $job, ?string $queue = null, ?string $connection = null): CallbackEvent
```

Comment:
I thought it will be better to make the signature properly typed as it can be done with php8.2. On one hand this is a BC for those extending this class. On another hand this is internal class and should not be extended.
Let me know what do you think. In case you consider this is as a major BC break, I would revert back signature as don't want to wait v12 to get this fix.